DependencyInjection/DoctrineExtension.php

```diff
@@ -199,6 +200,14 @@ protected function getConnectionOptions($connection)
     {
         $options = $connection;
 
+        if (! isset($options['charset'])) {
+            if (isset($options['url'])) {
+                $options['charset'] = strpos($options['url'], 'mysql') === 0 ? 'utf8mb4' : 'utf8';
```

Comment:
the scheme might also be `pdo_mysql` or `mysqli` as that's the name of the DBAL drivers, which are also supported as schemes.

Reply:
The scheme mysqli also starts with mysql, so that should be fine.

I didn't account for pdo_mysql, I will look into that.

Reply:
I don't think pdo_mysql can be use as a scheme, parse_url doesn't work with it. So that leaves just mysql, mysql2 and mysqli as a scheme.
```
$url = 'pdo_mysql://dbuser:dbpassword@127.0.0.1:3306/dbname?charset=utf8';
var_dump(parse_url($url));

array(2) {
  ["path"]=>  string(51) "pdo_mysql://dbuser:dbpassword@127.0.0.1:3306/dbname"
  ["query"]=>  string(12) "charset=utf8"
}
```